### ISSUE TITLE
dom-repeat: Duplicate values result in unsynchronized data

### ISSUE THREAD

**Issue by kevinpschaaf** [Polymer/polymer#1913]
The collection-based key abstraction currently assumes unique array element identity.

Problem can be observed here (type `bbbbb` in the top input, and observe data out of sync):
http://jsbin.com/rabize/edit?html,output

**Comment by zmoshansky** [comment]
+1, this would be really nice to have.
The workaround as suggested https://github.com/Polymer/polymer/issues/2667#issuecomment-167875621 is to wrap primitives in an object. 
`['analgesic', null, 'analgesic' ]` -> `[{k: 'analgesic'}, {k: null}, {k: 'analgesic'}]`

**Comment by PSG-Luna** [comment]
I've come across this issue as well, it's quite a big problem when having an array of any non object type

**Comment by rwestlund** [comment]
I just ran into this with 1.4.0, but I don't believe I have duplicate values in my string array.

I have this tap handler:
```
addStep: function() {
    this.push("recipe.directions", "");
}
```
and I'm using paper-inputs for each `recipe.directions` element within dom-repeat.  I'm adding text to each step before I add the next one, so I don't think there are ever duplicate values.

I found the following workaround:
```
this.push("recipe.directions", new String());
```

This likely works because `typeof new String() === 'object'`
